shared_model/interfaces/iroha_internal/transaction_batch_factory_impl.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "interfaces/iroha_internal/transaction_batch_factory_impl.hpp"
+
+#include "interfaces/iroha_internal/transaction_batch_impl.hpp"
+#include "interfaces/transaction.hpp"
+#include "validators/answer.hpp"
+
+shared_model::interface::TransactionBatchFactoryImpl::FactoryImplResult
+shared_model::interface::TransactionBatchFactoryImpl::createTransactionBatch(
+    const types::SharedTxsCollectionType &transactions) const {
+  std::string reason_name = "Transaction batch: ";
+  validation::ReasonsGroupType batch_reason;
+  batch_reason.first = reason_name;
+
+  if (boost::empty(transactions)) {
+    batch_reason.second.emplace_back("Provided transactions list is empty");
+  }
+
+  bool has_at_least_one_signature =
+      std::any_of(transactions.begin(), transactions.end(), [](const auto tx) {
+        return not boost::empty(tx->signatures());
+      });
+  if (not has_at_least_one_signature) {
+    batch_reason.second.emplace_back(
+        "Transaction batch should contain at least one signature");
+  }
+
+  if (not batch_reason.second.empty()) {
```

Comment:
It seems that validation that was before (`TransactionBatchFactory::createTransactionBatch`) is gone. Could you explain where is it moved?

Reply:
Which exactly validation? Most its part was moved into `transaction_sequence_factory`, though a couple of cases are still left here